Model/Data/Product/DataValidator.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Itonomy\Katanapim\Model\Data\Product;
+
+use Itonomy\Katanapim\Model\Data\Product\DataValidator\ValidatorInterface;
+
+class DataValidator
+{
+    /**
+     * @var ValidatorInterface[]
+     */
+    private array $defaultValidators;
+
+    /**
+     * @var ValidatorInterface[]
+     */
+    private array $scopeValidators;
+
+    /**
+     * @var array
+     */
+    private array $validators;
+
+    /**
+     * DataValidator constructor.
+     *
+     * @param array $validators
+     */
+    public function __construct(
+        array $validators
+    ) {
+        $this->validators = $validators;
+    }
+
+    /**
+     * Validate data.
+     *
+     * @param array $data
+     * @param bool $scope
+     * @return array
+     */
+    public function execute(array &$data, bool $scope = false): array
```

Comment:
Maybe im missing something but it looks to me like it will try to import url keys in glabal scope without validating them.

Reply:
Problem for url_keys exist on scope store import, the initial import in global scope will be handled correctly by Magento itself

Reply:
Then why add the validation to global scope?